### DISCUSSION TITLE
Error communicating with OpenAI

### DISCUSSION THREAD

**Discussion by desurdesur** [n3d1117/chatgpt-telegram-bot#250] [Q&A]
Hi there!
Did you meet the auth problem with OpenAI?

I've download and setup locally and catch the error when i try to send smth to my Telegram bot
```
Failed to get response ⚠️ An error has occurred. ⚠️
Error communicating with OpenAI
```

I've already:
- double checked OpenAI token
- double checked my OpenAI balance
- tested to send messages to OpenAI by simple python code with that OpenAI token – it works!
- asked ChatGPT how to fix it! 

Any ideas how to fix it?

**Comment by n3d1117** [comment]
Hey there, can you share your .env file? (redact token and api key)

Also if you run the bot via pythonanywhere.com, you'll get this error as they have restrictions on outbound traffic

**Reply by desurdesur** [reply]
Thanks for your promt reply! 

My .env file is
```
# Your OpenAI API key
OPENAI_API_KEY='sk-XXX'

# Your Telegram bot token obtained using @BotFather
TELEGRAM_BOT_TOKEN='XXX'

# Telegram user ID of admins, or - to assign no admin
ADMIN_USER_IDS='-'

# Comma separated list of telegram user IDs, or * to allow all
ALLOWED_TELEGRAM_USER_IDS='*'
```

I don't use pythonanywhere.com or so. Trying to launch it from my local machine at first.

**Reply by n3d1117** [reply]
Could you try removing all `'` characters from the .env file?

**Reply by desurdesur** [reply]
Unfortunately didn't help

Anyway I've write a simple bot – it works. So it's no hurry. And if you have any ideas what it might be a auth issue – please share.

Thanks

**Reply by ivankoriako** [reply]
Did you fix it?

**Reply by getmanele** [reply]
# Your OpenAI API key
# OPENAI_API_KEY=sk-my-sk2
OPENAI_API_KEY=sk-my-sk

# Your Telegram bot token obtained using @BotFather
TELEGRAM_BOT_TOKEN=my_token

# Telegram user ID of admins, or - to assign no admin
ADMIN_USER_IDS=265815***,

# Comma separated list of telegram user IDs, or * to allow all
ALLOWED_TELEGRAM_USER_IDS=*

# Optional configuration, refer to the README for more details
BUDGET_PERIOD=monthly
USER_BUDGETS=100.0
# GUEST_BUDGET=100.0
# TOKEN_PRICE=0.002
# IMAGE_PRICES=0.016,0.018,0.02
# TRANSCRIPTION_PRICE=0.006
# ENABLE_QUOTING=true
ENABLE_IMAGE_GENERATION=true
ENABLE_TRANSCRIPTION=true
# PROXY=http://localhost:8080
OPENAI_MODEL=gpt-3.5-turbo
# ASSISTANT_PROMPT="You are a helpful assistant."
# SHOW_USAGE=false
# STREAM=true
# MAX_TOKENS=1200
# MAX_HISTORY_SIZE=15
# MAX_CONVERSATION_AGE_MINUTES=180
# VOICE_REPLY_WITH_TRANSCRIPT_ONLY=true
# VOICE_REPLY_PROMPTS="Hi bot;Hey bot;Hi chat;Hey chat; Hey Dobby"
# N_CHOICES=1
# TEMPERATURE=1.0
# PRESENCE_PENALTY=0.0
# FREQUENCY_PENALTY=0.0
# IMAGE_SIZE=512x512
GROUP_TRIGGER_KEYWORD="Добби"
# IGNORE_GROUP_TRANSCRIPTIONS=true
BOT_LANGUAGE=ru

**Comment by Nail-Al** [comment]
Have the same issue by running locally w/ PyCharm & console.

here is the .env
```
# Your OpenAI API key
OPENAI_API_KEY=sk-myapikey

# Your Telegram bot token obtained using @BotFather
TELEGRAM_BOT_TOKEN=mybotactualtoken

# Telegram user ID of admins, or - to assign no admin
ADMIN_USER_IDS=xmyactualIDx

# Comma separated list of telegram user IDs, or * to allow all
ALLOWED_TELEGRAM_USER_IDS=

# Optional configuration, refer to the README for more details
# BUDGET_PERIOD=monthly
# USER_BUDGETS=*
# GUEST_BUDGET=100.0
# TOKEN_PRICE=0.002
# IMAGE_PRICES=0.016,0.018,0.02
# TRANSCRIPTION_PRICE=0.006
# ENABLE_QUOTING=true
# ENABLE_IMAGE_GENERATION=true
# ENABLE_TRANSCRIPTION=true
# PROXY=http://localhost:8080
OPENAI_MODEL=gpt-3.5-turbo
# ASSISTANT_PROMPT="You are a helpful assistant."
# SHOW_USAGE=false
# STREAM=true
MAX_TOKENS=1200
MAX_HISTORY_SIZE=15
# MAX_CONVERSATION_AGE_MINUTES=180
# VOICE_REPLY_WITH_TRANSCRIPT_ONLY=true
# VOICE_REPLY_PROMPTS="Hi bot;Hey bot;Hi chat;Hey chat"
# N_CHOICES=1
# TEMPERATURE=1.0
# PRESENCE_PENALTY=0.0
# FREQUENCY_PENALTY=0.0
# IMAGE_SIZE=512x512
# GROUP_TRIGGER_KEYWORD=""
# IGNORE_GROUP_TRANSCRIPTIONS=true
# BOT_LANGUAGE=en
```

error:

```
  File ".../openai_helper.py", line 257, in __common_get_chat_response
    raise Exception(f"⚠️ _{localized_text('error', bot_language)}._ ⚠️\n{str(e)}") from e
Exception: ⚠️ _An error has occurred._ ⚠️
Error communicating with OpenAI

2023-10-24 14:14:27,777 - telegram.ext.Updater - ERROR - Error while getting Updates: Conflict: terminated by other getUpdates request; make sure that only one bot instance is running
2023-10-24 14:14:27,778 - root - ERROR - Exception while handling an update: Conflict: terminated by other getUpdates request; make sure that only one bot instance is running


```

**Reply by ismailosmnv** [reply]
Did you fix it?

**Reply by koren4ik** [reply]
i have the same error. do u fix it?

**Reply by Nail-Al** [reply]
Just went with Docker version - it's working for me 👍🏼

**Reply by ismailosmnv** [reply]
https://github.com/openai/openai-python/issues/56

**Comment by Ele1ment** [comment]
Failed to get response ⚠️ An error has occurred. ⚠️
'ascii' codec can't encode characters in position 7-9: ordinal not in range(128)
